nitro-protocol/contracts/LedgerFinancingApp.sol

```diff
@@ -0,0 +1,151 @@
+    //  - the latest consensus principal
+    //  - the channel's interest rate
+    //  - the time elapsed since the last principal adjustment
+    function getOutstandingInterest(InterestAppData memory appData)
+        private
+        view
+        returns (Funds memory)
+    {
+        uint32 numDays = daysSince(appData.blocknumber);
+        uint256 simpleInterestNum = numDays * appData.dpyNum;
+
+        address[] memory assets = new address[](appData.principal.asset.length);
+        uint256[] memory amounts = new uint256[](appData.principal.asset.length);
+
+        Funds memory outstanding = Funds(assets, amounts);
+
+        // copy all assets from the principal, and multiply by the interest rate
+        for (uint256 i = 0; i < appData.principal.asset.length; i++) {
+            outstanding.asset[i] = appData.principal.asset[i];
+            outstanding.amount[i] =
+                (appData.principal.amount[i] * simpleInterestNum) /
+                appData.dpyDen;
+        }
+
+        return outstanding;
+    }
+
+    // Ensures that the given outcome does not unfairly allocate to the intermediary.
+    function requireOutcomeIsEarned(
+        Outcome.SingleAssetExit[] memory initialOutcome,
+        Outcome.SingleAssetExit[] memory finalOutcome,
+        Funds memory outstandingInterest
+    ) private pure {
+        for (uint256 i = 0; i < outstandingInterest.asset.length; i++) {
+            address asset = outstandingInterest.asset[i];
+            uint256 earned = outstandingInterest.amount[i];
+
+            for (uint256 j = 0; j < finalOutcome.length; j++) {
+                if (finalOutcome[j].asset == asset) {
+                    require(initialOutcome[j].asset == asset, 'Asset mismatch');
+                    require(
+                        initialOutcome[j]
+                            .allocations[uint256(AllocationIndicies.intermediary)]
+                            .destination ==
+                            finalOutcome[j]
+                                .allocations[uint256(AllocationIndicies.intermediary)]
+                                .destination,
+                        'payee mismatch'
+                    );
+                    // combine prior balance with outstandingInterest
+                    earned += initialOutcome[j]
+                        .allocations[uint256(AllocationIndicies.intermediary)]
+                        .amount;
+                    // and compare against the claimed outcome
+                    uint256 claimed = finalOutcome[j]
+                        .allocations[uint256(AllocationIndicies.intermediary)]
+                        .amount;
+
+                    require(claimed <= earned, 'earned<claimed');
+                }
+            }
+        }
+    }
+
+    function daysSince(uint256 blocknumber) private view returns (uint32) {
+        return uint32((block.number - blocknumber) / 7200); // 7200 == 24*60*60/12
+    }
```

Comment:
Curious about the best way to convert from block number to time.  Is using 12 second block time is potentially a bit risky? Since The Merge, Ethereum mainnet will have a 12 second blocktime much more reliably... but PoW chains as well as say Filecoin Mainnet may have different periods and its not impossible for them to vary over time.

There is also `block.timestamp` available (which may have its own risks).

(Side note -- good thing we did #1036 recently!)

I wonder if it makes the most sense to define a yield per block, rather than per unit time? Then, participants may adjust this off-chain in light of a change in blocktime (from switching networks, say). 

Of course then interest rates will be "smaller" (bigger denominators) but I don't think this makes integer arithmetic more risky in the end. In fact, a downside to how you are doing it here is to divide the number of elapsed blocks by 7200 _before_ you multiply by the interest numerator and principal amount. I think you want to do all of your multiplications up front and only divide after that? (I see some evidence you had that in mind above anyway).

Reply:
Yield per block implemented: [353e56b](https://github.com/statechannels/go-nitro/pull/1107/commits/353e56b4a179eb38fa9c9e7281e14eb392285789)